### ISSUE TITLE
linux-vm instances should have consistent Chrome version

### ISSUE THREAD

**Issue by yjbanov** [flutter/flutter#54147]
I followed instructions for creating a new linux VM instance: https://g3doc.corp.google.com/company/teams/flutter/devicelab/linux_host.md?cl=head#new-virtual-host

However, there's no mention of Chrome and the `devicelab-linux-image` does not come with a preinstalled version.

We need to make sure there's a consistent version of Chrome installed on all linux VM instances.

I'm not sure what's the right approach is. Some possibilities:

* Maintain a version lock in the source, install and cache on-demand (this is what flutter/engine does, although there we use Chromium, not Chrome). The benefit of that approach is that it's easy to change the version of Chrome, including rolling it back.
* Have it preinstalled as part of `devicelab-linux-image`.

**Comment by godofredoc** [comment]
The optimal way to provide consistent versions for browsers is to bundle them in cipd packages. Then both device lab and luci can take advantage of it.

**Comment by nturgut** [comment]
I'm planning to carry the chrome download code to [web_installers](https://github.com/flutter/web_installers) We can use this form the recipe to download/install chrome.

**Comment by godofredoc** [comment]
We definitely can download and install but the builds may not be hermetic and we won't be using the caching mechanisms provided by luci.

Some benefits of providing browsers as cipd packages:

 * LUCI automatically downloads and caches cipd packages.
 * We can keep different versions and request them by tags.
 * Builds will keep references to the version of the browser used to test. This will be required for managing branching.
 * Additionally we can use as much as possible portable binaries to keep our bots as hermetic as possible.

**Comment by nturgut** [comment]
If I convert the web_installers code to runnable that can be used inside a recipe, for example:
```
dart browser_installer.dart --browser=chrome --version=76.0
```

Is this useful for using it in combination with cipd API?

**Comment by godofredoc** [comment]
Does that mean the tools will still be in charge of downloading the the binaries?

cipd is package management system that consists of a service where the package is stored and cli to interact with the service.

From recipes you can download the dependencies using:

 * adding it the the builder configuration - https://flutter.googlesource.com/recipes/+/refs/heads/master/recipes/engine.py#751 

Using a configuration file we can translate it to something like:

```
task:
    name: web_engine_firefox
    cipd_deps:
         path: flutter/infra/browsers/firefox
          tag: version_75
```

Then the test can just assume the correct version of firefox will be available

**Comment by nturgut** [comment]
Yes, the we can use a tool writen by us to download the binaries such as browsers and browser drivers. I added more information here: https://github.com/flutter/flutter/issues/56058#issuecomment-622123219
